proposals/interface-types/working-notes/choices/webidl.md

```diff
@@ -0,0 +1,270 @@
+array buffers), allowing sharing on values contradicts one of the key design
+criteria for Interface Types -- namely supporting interoperability in a cross
+ownership domain where there is no sharing.
+
+In practice, Web APIs that are specified using WebIDL make heavy use of extended
+attributes to convey additional semantics that are very JavaScript centric. Such
+usages would have to be prohibited when designing APIs for languages other than
+JavaScript.
+
+This again leads to a situation where the 'real' IDL that would be usable for
+WebAssembly would be a limited form of a public IDL. This is not an ideal model
+for standardization nor for supporting the wider WebAssembly ecosystem.
+
+### Nominal Types
+
+JavaScript does not have a static model of types. On the other hand, many
+programming languages do have static types. One of the features of many static type systems is the so-called _nominal type_.
+
+A nominal type differs from a structural type in several ways; but one of the
+most fundamental ways is that a nominal type can be used to model entities that
+are not directly conveyed by the actual data used to denote them.
+
+For example, one may choose to model a chair using the nominal type `Chair`;
+whose values contain a pair of integers: the SKU and price (for example). The
+pair of integers is _not equivalent_ to a chair, but represent the information
+needed to model chairs in some application or API. 
+
+A different entity (a person say) may also be modeled by a type `Person` which
+contains two integers (for example, an indentification number and an index into
+a street address table).
+
+A given pair of integers might be either a `Chair` or a `Person` (or a 2D point)
+and there is no way of conveying the distinction unless the nominal type is also
+taken into consideration.
+
+### Algebraic Type Definitions
+
+The Interface Types language permits the definition of a limited form of
+_algebraic type definition_. This is, in part, to regularize concepts such as
+nullability and to permit limited forms of polymorphism (allowing a color to be
+named by a string, a vector of small integers or a large integer, for example).
+
+WebIDL explicitly prohibits the formation of new types; indeed it uses type
+unions to achieve similar polymorphism. However, as we note above, not all
+programming languages support dynamic types or type unions; which makes it
+problematic to support such polymorphism.
+
+The form of algebraic types envisaged in the Interface Types system is
+intentionally limited. For example, we do not permit _recursive_ types to be
+defined. This is in keeping with the vision of supporting APIs across ownership
+boundaries where complex data is likely to take limited form.[^This design
+choice may be revisited in future versions of the Interface Types specification
+if supporting structured types such as abstract XML and/or JSON become
+important.]
```

Comment:
I'd guess this is also too much in the weeds unless the point you're making is that WebIDL *does* support cyclic ADTs (I think it can, b/c dictionaries are named and can thus create cycles?), and so this is another case of how we'd want to subset WebIDL for wasm's needs.